agent/api/statechange.go

```diff
@@ -23,14 +23,25 @@ import (
 	"github.com/aws/amazon-ecs-agent/agent/statechange"
 	"github.com/aws/amazon-ecs-agent/ecs-agent/api/attachment"
 	apicontainerstatus "github.com/aws/amazon-ecs-agent/ecs-agent/api/container/status"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/api/ecs"
+	ecsmodel "github.com/aws/amazon-ecs-agent/ecs-agent/api/ecs/model/ecs"
 	apitaskstatus "github.com/aws/amazon-ecs-agent/ecs-agent/api/task/status"
 	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger/field"
 	ni "github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/networkinterface"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/utils"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/docker/go-connections/nat"
 	"github.com/pkg/errors"
 )
 
+const (
+	// ecsMaxNetworkBindingsLength is the maximum length of the ecs.NetworkBindings list sent as part of the
+	// container state change payload. Currently, this is enforced only when containerPortRanges are requested.
+	ecsMaxNetworkBindingsLength = 100
```

Comment:
is there any chance that this might be a behavior change for a customer?  
Is the 100 max limit already enforced in the control plane, or is this the first that we're enforcing [what our docs say](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_NetworkBinding.html)?

Reply:
> is there any chance that this might be a behavior change for a customer?

This is not a behavior change for a customer but rather persisting the current behavior, see current code: https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/ecsclient/client.go#L58-L60

> Is the 100 max limit already enforced in the control plane, or is this the first that we're enforcing [what our docs say](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_NetworkBinding.html)?

The limit was originally introduced as part of this pull request: https://github.com/aws/amazon-ecs-agent/pull/3479

I synced up with Anuj (the original code author of the above) who mentioned that, as of the time the above was implemented, control plane is enforcing the 100 max limit and that Agent is set to fail early in this case. We could remove this existing behavior on our Agent if we wanted to, but I think the drawback of that is that Agent may unnecessarily send container state changes to control plane that will result in failure.